allennlp/semparse/contexts/table_question_context.py

```diff
@@ -0,0 +1,307 @@
+             "further", "ourselves", "out", "what", "for", "weren", "does", "above", "between", "mustn", "?",\
+             "be", "hasn", "who", "were", "here", "shouldn", "let", "hers", "by", "both", "about", "couldn",\
+             "of", "could", "against", "isn", "or", "own", "into", "while", "whom", "down", "wasn", "your",\
+             "from", "her", "their", "aren", "there", "been", ".", "few", "too", "wouldn", "themselves",\
+             ":", "was", "until", "more", "himself", "on", "but", "don", "herself", "haven", "those", "he",\
+             "me", "myself", "these", "up", ";", "below", "'re", "can", "theirs", "my", "and", "would", "then",\
+             "is", "am", "it", "doesn", "an", "as", "itself", "at", "have", "in", "any", "if", "!", "again", "'ll",\
+             "no", "that", "when", "same", "how", "other", "which", "you", "many", "shan", "'t", "'s", "our",
+  "after","most", "'d", "such", "'m", "why", "a", "off", "i", "yours", "so", "the", "having", "once"]
+
+DEFAULT_NUMBERS = ['-1', '0', '1']
+NUMBER_CHARACTERS = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '.', '-'}
+MONTH_NUMBERS = {
+        'january': 1,
+        'jan': 1,
+        'february': 2,
+        'feb': 2,
+        'march': 3,
+        'mar': 3,
+        'april': 4,
+        'apr': 4,
+        'may': 5,
+        'june': 6,
+        'jun': 6,
+        'july': 7,
+        'jul': 7,
+        'august': 8,
+        'aug': 8,
+        'september': 9,
+        'sep': 9,
+        'october': 10,
+        'oct': 10,
+        'november': 11,
+        'nov': 11,
+        'december': 12,
+        'dec': 12,
+        }
+ORDER_OF_MAGNITUDE_WORDS = {'hundred': 100, 'thousand': 1000, 'million': 1000000}
+NUMBER_WORDS = {
+        'zero': 0,
+        'one': 1,
+        'two': 2,
+        'three': 3,
+        'four': 4,
+        'five': 5,
+        'six': 6,
+        'seven': 7,
+        'eight': 8,
+        'nine': 9,
+        'ten': 10,
+        'first': 1,
+        'second': 2,
+        'third': 3,
+        'fourth': 4,
+        'fifth': 5,
+        'sixth': 6,
+        'seventh': 7,
+        'eighth': 8,
+        'ninth': 9,
+        'tenth': 10,
+        **MONTH_NUMBERS,
+        }
+
+
+class TableQuestionContext:
+    """
+    A Barebones implementation similar to https://github.com/crazydonkey200/neural-symbolic-machines/blob/master/table/wtq/preprocess.py
+    for extracting entities from a question given a table and type its columns with <string> | <date> | <number>
+    """
+
+    def __init__(self,
+                 cell_values: Set[str],
+                 column_type_statistics: List[Dict[str, int]],
+                 column_index_to_name:  Dict[int, str]) -> None:
+        self.cell_values = cell_values
+        self.column_types = {column_index_to_name[column_index] : max(column_type_statistics[column_index], 
+                                           key=column_type_statistics[column_index].get) for column_index in column_index_to_name}
```

Comment:
Can you also add another field to this class that stores all the content in the table as a `dict` so that we can pass it directly to the executor? See [here](https://github.com/allenai/allennlp/blob/2dd05bdbe38f2c0ab2c23a7bd06d238b830791eb/allennlp/semparse/executors/wikitables_variable_free_executor.py#L71) for the format expected by the executor.

Reply:
Also, I think `question_tokens` should be a field in this class. Then `get_entities_from_question` need not take any arguments.